File: Application/app/src/main/java/com/example/application/FoodPrefActivity.java
package com.example.application;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.Switch;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;
import java.util.HashMap;

public class FoodPrefActivity extends AppCompatActivity {

    private EditText numberofRecipes;
    private Switch maximizeIngredients;
    private Switch ignorePantry;
    private ArrayList<String> arrayList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main2);

        Button button_next_API = findViewById(R.id.button_next_API);

        numberofRecipes = findViewById(R.id.number_of_recipes);
        maximizeIngredients = findViewById(R.id.maximizeIngredients);
        ignorePantry = findViewById(R.id.ignorePantry);
        arrayList = getIntent().getStringArrayListExtra("foods");

        HashMap<Integer, Object> preferences = new HashMap<>();

        // pass data
        button_next_API.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                passDataToApiTime();
            }
        });
    }

    private void passDataToApiTime() {
        String numberOfRecipes = this.numberofRecipes.getText().toString();
        Boolean maximizeIngredients = this.maximizeIngredients.isChecked();
        Boolean ignorePantry = this.ignorePantry.isChecked();

        HashMap preferences = new HashMap();
        preferences.put("numberOfRecipes", numberOfRecipes);
        preferences.put("maximizeIngredients", maximizeIngredients);
        preferences.put("ignorePantry", ignorePantry);

        Intent intent = new Intent(this, ApiTime.class);
        intent.putExtra("foods", arrayList);
        intent.putExtra("preferences", preferences);
        startActivity(intent);

        startActivity(intent);
    }
}

File: Application/app/src/main/java/com/example/application/FoodPrefActivity.kt
package com.example.application

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.widget.Button
import android.widget.EditText
import android.widget.Switch
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity

class FoodPrefActivity : AppCompatActivity() {
    private var numberofRecipes: EditText? = null
    private var maximizeIngredients: Switch? = null
    private var ignorePantry: Switch? = null
    private var arrayList: ArrayList<String>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.enableEdgeToEdge()
        setContentView(R.layout.activity_main2)

        val button_next_API = findViewById<Button>(R.id.button_next_API)
        val button_previous = findViewById<Button>(R.id.button_prev_foodPref)

        numberofRecipes = findViewById(R.id.number_of_recipes)
        maximizeIngredients = findViewById(R.id.maximizeIngredients)
        ignorePantry = findViewById(R.id.ignorePantry)
        arrayList = intent.getStringArrayListExtra("foods")

        // pass data
        button_next_API.setOnClickListener { passDataToApiTime() }

        button_previous.setOnClickListener {
            val intent = Intent(applicationContext, MainActivity::class.java)
            startActivity(intent)
        }
    }

    private fun passDataToApiTime() {
        val numberOfRecipes = numberofRecipes!!.text.toString()
        val maximizeIngredients = maximizeIngredients!!.isChecked
        val ignorePantry = ignorePantry!!.isChecked

        // Use HashMap with specific types
        val preferences: HashMap<String, Any> = HashMap()
        preferences["numberOfRecipes"] = numberOfRecipes
        preferences["maximizeIngredients"] = maximizeIngredients
        preferences["ignorePantry"] = ignorePantry

        setContentView(R.layout.activity_timeout)

        Handler(Looper.getMainLooper()).postDelayed({
            val intent = Intent(this, ApiTime::class.java)
            intent.putExtra("foods", arrayList)
            intent.putExtra("preferences", preferences)
            startActivity(intent)
        }, 3000)
    }

}
